roofit/roofit/inc/RooLagrangianMorphFunc.h

```diff
@@ -0,0 +1,363 @@
+  void setParameter(const char *name, double value, double min, double max,
+                    double error);
+  void randomizeParameters(double z);
+  const RooArgSet *getParameterSet() const;
+  ParamSet getMorphParameters(const char *foldername) const;
+  ParamSet getMorphParameters() const;
+
+  RooLagrangianMorphFunc *getLinear() const;
+
+  int nParameters() const;
+  int nPolynomials() const;
+
+  bool isCouplingUsed(const char *couplname);
+  const RooArgList *getCouplingSet() const;
+  ParamSet getCouplings() const;
+
+  // virtual Bool_t IsAllowNegativeYields() const { return
+  // this->_config.IsAllowNegativeYields(); }
+  TMatrixD getMatrix() const;
+  TMatrixD getInvertedMatrix() const;
+  double getCondition() const;
+
+  RooRealVar *getObservable() const;
+  RooRealVar *getBinWidth() const;
+
+  void printEvaluation() const;
+  void printCouplings() const;
+  void printFlags() const;
+  void printPhysics() const;
+
+  RooProduct *getSumElement(const char *name) const;
+
+  std::vector<std::string> getSamples() const;
+
+  double expectedUncertainty() const;
+  TH1 *createTH1(const std::string &name);
+  TH1 *createTH1(const std::string &name, bool correlateErrors);
+
+protected:
+  class CacheElem;
+  void init();
+  void printAuthors() const;
+  void setup(bool ownParams = true);
+  bool _ownParameters = false;
+
+  /* wip
+  void disableInterference(const std::vector<const char*>& nonInterfering) ;
+  void disableInterferences(const std::vector<std::vector<const char*> >&
+  nonInterfering) ;
+  */
+
+  mutable RooObjCacheManager _cacheMgr; //! The cache manager
+
+  void addFolders(const RooArgList &folders);
+
+  bool hasCache() const;
+  RooLagrangianMorphFunc::CacheElem *getCache(const RooArgSet *nset) const;
+  void updateSampleWeights();
+
+  RooRealVar *setupObservable(const char *obsname, TClass *mode,
+                              TObject *inputExample);
+
+public:
+  double implementedPrecision();
+  void importToWorkspace(RooWorkspace *ws, const RooAbsReal *object);
+  void importToWorkspace(RooWorkspace *ws, RooAbsData *object);
+
+  void writeMatrixToFile(const TMatrixD &matrix, const char *fname);
+  void writeMatrixToStream(const TMatrixD &matrix, std::ostream &stream);
+  TMatrixD readMatrixFromFile(const char *fname);
+  TMatrixD readMatrixFromStream(std::istream &stream);
+
+  // RooDataHist* makeDataHistogram(TH1* hist, RooRealVar* observable, const
+  // char* histname = NULL) ; void setDataHistogram(TH1* hist, RooRealVar*
+  // observable, RooDataHist* dh) ; void printDataHistogram(RooDataHist* hist,
+  // RooRealVar* obs) ;
+
+  int countSamples(std::vector<RooArgList *> &vertices);
+  int countSamples(int nprod, int ndec, int nboth);
+
+  TPair *makeCrosssectionContainer(double xs, double unc);
+  std::map<std::string, std::string>
+  createWeightStrings(const ParamMap &inputs,
+                      const std::vector<std::string> &couplings);
```

Comment:
```suggestion
```
I suggest to delete this function. See corresponding comment in the `.cxx` file.